packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue

```diff
@@ -0,0 +1,364 @@
+import { VueGlobalisationMixin } from '@justeat/f-globalisation';
+
+import ButtonComponent from '@justeat/f-button';
+import '@justeat/f-button/dist/f-button.css';
+
+import LegacyBanner from './LegacyBanner.vue';
+
+import tenantConfigs from '../tenants';
+
+export default {
+    name: 'CookieBanner',
+
+    components: {
+        ButtonComponent,
+        LegacyBanner
+    },
+
+    mixins: [VueGlobalisationMixin],
+
+    props: {
+
+        locale: {
+            type: String,
+            default: 'en-GB'
+        },
+
+        isHidden: {
+            type: Boolean,
+            default: false
+        },
+
+        showLegacyBanner: {
+            type: Boolean,
+            default: false
+        },
+
+        cookieExpiry: {
+            type: Number,
+            default: 7776000
+        }
+    },
+
+    data () {
+        const locale = globalisationServices.getLocale(tenantConfigs, this.locale, this.$i18n);
+        const localeConfig = tenantConfigs[locale];
+        const theme = globalisationServices.getTheme(locale);
+
+        return {
+            config: { ...localeConfig },
+            tenantConfigs,
+            theme,
+            hideBanner: false,
+            legacyBanner: false
+        };
+    },
+
+    watch: {
+        isHidden (newVal) {
+            this.hideBanner = !!newVal;
+        },
+
+        showLegacyBanner (newVal) {
+            this.legacyBanner = !!newVal;
+        }
+    },
+
+    mounted () {
+        this.checkLegacyBannerFlag();
+        this.checkCookieBannerCookie();
+        this.focusOnTitle();
+    },
+
+    methods: {
+        /**
+         * Actions for "Accept all cookies" button
+         */
+        acceptActions () {
+            this.setCookieBannerCookie('full');
+            this.setLegacyCookieBannerCookie();
+            this.dataLayerPush('full');
+            this.hideBanner = true;
+        },
+        /**
+         * Actions for "Accept only required cookies" button
+         */
+        nonAcceptActions () {
+            this.setCookieBannerCookie('necessary');
+            this.setLegacyCookieBannerCookie();
+            this.dataLayerPush('necessary');
+            this.resendEvents();
+            this.removeUnnecessaryCookies();
+            this.hideBanner = true;
+        },
+        /**
+         * Set focus to the cookie consent banner title for accessibility
+         */
+        focusOnTitle () {
+            this.$refs.cookieBannerHeadingFocus.focus();
+        },
+        /**
+         * Check if the cookie banner has been shown to this user
+         */
+        checkLegacyBannerFlag () {
+            this.legacyBanner = this.config.displayLegacy;
+        },
+        /**
+         * Check if the cookie banner has been shown to this user
+         */
+        checkCookieBannerCookie () {
+            if (this.legacyBanner) {
+                this.hideBanner = this.$cookies.get('je-banner_cookie') === 2;
```

Comment:
I think we should extract these cookie names into constants.

Reply:
Done